src/index.js

```diff
@@ -27,25 +27,29 @@ export function rangeFromLineNumber(textEditor: TextEditor, line: number, column
     throw new Error(`Line number (${lineNumber}) greater than maximum line (${lineMax})`)
   }
 
-  let colStart = column
-  if (!Number.isFinite(colStart) || Number.isNaN(colStart) || colStart < 0) {
-    const indentation = buffer.lineForRow(lineNumber).match(/^\s+/)
-    if (indentation && indentation.length) {
-      colStart = indentation[0].length
-    } else {
-      colStart = 0
-    }
+  const columnGiven = typeof column === 'number' && Number.isFinite(column) && column > -1
+  let lineText = buffer.lineForRow(lineNumber)
+  const lineLength = lineText.length
+  if (columnGiven && column > lineLength) {
+    throw new Error(`Column start (${column}) greater than line length (${lineLength})`)
+  } else if (columnGiven) {
+    lineText = lineText.substr(column)
   }
 
-  const lineLength = buffer.lineLengthForRow(lineNumber)
-
-  if (colStart > lineLength) {
-    throw new Error(`Column start (${colStart}) greater than line length (${lineLength})`)
+  let colEnd = lineLength
+  let colStart = columnGiven ? column : 0
+  const rowText = buffer.lineForRow(lineNumber).substr(colStart)
```

Comment:
You're duplicating the logic from lines 31-36... aren't the lines below meant to just use `lineText`?